nextcord/emoji.py

```diff
@@ -212,29 +245,55 @@ async def edit(
             The new emoji name.
         roles: Optional[List[:class:`~nextcord.abc.Snowflake`]]
             A list of roles that can use this emoji. An empty list can be passed to make it available to everyone.
+
+            This is only available when editing an emoji in a guild.
         reason: Optional[:class:`str`]
             The reason for editing this emoji. Shows up on the audit log.
 
+            This is only available when editing an emoji in a guild.
+
         Raises
         ------
         Forbidden
             You are not allowed to edit emojis.
         HTTPException
             An error occurred editing the emoji.
+        ValueError
+            The emoji does not belong to a guild or application
 
         Returns
         -------
         :class:`Emoji`
             The newly updated emoji.
         """
+        if self.guild is None and self.application_id is None:
+            raise ValueError("Cannot edit an emoji without a guild_id or application_id.")
```

Comment:
checking/raising twice? (also on line 291)

Reply:
The second one is for type checking reasons, as it doesn't always return an Emoji, according to it.

Reply:
makes sense 👍